static/app/stores/projectsStore.tsx

```diff
@@ -158,38 +157,33 @@ const storeConfig: ProjectsStoreDefinition = {
     }
 
     const newProject = {...project, teams: [...project.teams, team]};
+    this.projects = this.projects.map(p => (p.id === project.id ? newProject : p));
 
-    this.itemsById = {...this.itemsById, [project.id]: newProject};
     this.trigger(new Set([project.id]));
   },
 
   // Internal method, does not trigger
   removeTeamFromProject(teamSlug: string, project: Project) {
     const newTeams = project.teams.filter(({slug}) => slug !== teamSlug);
     const newProject = {...project, teams: newTeams};
-
-    this.itemsById = {...this.itemsById, [project.id]: newProject};
+    this.projects = this.projects.map(p => (p.id === project.id ? newProject : p));
   },
 
   isLoading() {
     return this.loading;
   },
 
-  getAll() {
-    return Object.values(this.itemsById).sort((a, b) => a.slug.localeCompare(b.slug));
-  },
```

Comment:
getAll here is creating a new array and sorting it. this is used in getState, getById, getBySlug